### DISCUSSION TITLE
I am curious if the type of listeners includes parameters for version compatibility reasons.

### DISCUSSION THREAD

**Discussion by MyohanMyolang** [pmndrs/zustand#2593] [Q&A] [answered]
When using useSyncExternal, they call subscribe like this:

```typescript
  const handleStoreChange = () => {
    if (checkIfSnapshotChanged(inst)) {
      forceStoreRerender(fiber);
    }
  };
  return subscribe(handleStoreChange);
```

However, the current Listener type is defined as:
```typescript
type Listner = (state: TState, prevState: TState) => void
```

Is this done for version compatibility reasons?

**Comment by dai-shi** [comment]
Are you referring to the difference between useSyncExternalStore listener and Zustand listener?

**Reply by MyohanMyolang** [reply]
umm maybe that's correct, they call what created in createStoreImpl `subscribe` then get `handleStoreChange` and add `listners`,
useSyncExternalStore gave no paramater function but listner type has 2 arguments state, and prevState
so i curious can it call listner, like this:
```typescript
type Listner = () => void
const listners: Set<Listner> = new Set()

const setState: StoreApi<TState>['setState'] = (partial, replace) => {
  const nextState =
    typeof partial === 'function'
      ? (partial as (state: TState) => TState)(state)
      : partial
  if (!Object.is(nextState, state)) {
    state =
      replace ?? (typeof nextState !== 'object' || nextState === null)
        ? (nextState as TState)
        : Object.assign({}, state, nextState)
    listeners.forEach((listener) => listener())
  }
}
```

**Reply by dai-shi** [reply, marked as answer]
Sure, that's technically possible. It's the design choice that Zustand listener takes `state` and `prevState`.

**Reply by MyohanMyolang** [reply]
thank you for answer!
